### PR TITLE
Move storage code into a subpackage

### PR DESCRIPTION
**Pull request by dhermes**
This PR was created via: https://gist.github.com/dhermes/e239691aa584bd56a5352e34aad27cf3

```
export PROJECT_DIR="${HOME}/google-cloud-python"
export READMES_DIR="${HOME}/i-wrote-some-readmes-for-2357"
cd ${PROJECT_DIR}
git worktree add -b make-storage-subpackage ../hotfix official/master
python make_commits.py \
  --git-root "${PROJECT_DIR}/../hotfix" \
  --package storage \
  --package-name "Google Cloud Storage" \
  --readme "${READMES_DIR}/storage/README.rst"
```

### CODE REVIEW

File: storage/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+.. _official Google Cloud Storage documentation: https://cloud.google.com/storage/docs/cloud-console#_creatingbuckets
+
+.. code:: python
+
+    from google.cloud import storage
+    client = storage.Client()
+    bucket = client.get_bucket('bucket-id-here')
+    # Then do other things...
+    blob = bucket.get_blob('remote/path/to/file.txt')
+    print blob.download_as_string()
```
Comment (jonparrott) [inline storage/README.rst]: `print(...)`?

File: tox.ini [hunk 2]
```diff
@@ -18,6 +19,12 @@ covercmd =
       --cov-append \
       --cov-config {toxinidir}/.coveragerc \
       core/unit_tests
+    py.test --quiet \
```
Comment (jonparrott) [inline tox.ini]: I'm not sure I understand why in `.travis.yml` you do `(cd storage && tox -e cover)` but you also do coverage for storage here. Does it not end up running the same tests twice?

Comment (dhermes) [inline tox.ini]: @jonparrott See https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2367#discussion_r79890514

File: storage/README.rst (hunk 1 above)
Comment (dhermes) [inline storage/README.rst]: Folded into the 2nd commit

File: tox.ini (hunk 2 above)
Comment (jonparrott) [inline tox.ini]: Resolved over chat. I'm cool with this.

**Comment by jonparrott** [comment]
LGTM

**Comment by tseaver** [comment]
- Won't the chnages to top-level files conflict with #2410?
- The duplication across the sub-repositories (`.coveragerc`, `MANIFEST.in`, etc.) is getting worse:  to we have an issue to track stripping that out?
- Likewise the duplication inside the top-level stuff (`.travis.yml`, `tox.ini`).

**Comment by dhermes** [comment]
@tseaver Yes essentially all will conflict, but luckily I can just re-run the script (that is why I ran it).

- The duplication of `.coveragerc` can be wrapped into a single file (put it in #2404 to track)
- The duplication of `MANIFEST.in` is essentially necessary, just like do-nothing `__init__.py` files are. Also note that over time these files may take on a life of their own
- The top-level stuff in (`tox.ini` and `.travis.yml`) will be greatly reduced when I address #2277

**Comment by tseaver** [comment]
OK, LGTM

**Comment by dhermes** [comment]
Pushed to resolve rebase conflict and re-ran script. Won't wait on Travis since a) that could be awhile and b) this will be painful.

**Pull request merged**